play.pokemonshowdown.com/js/client-teambuilder.js

```diff
@@ -1792,31 +1797,76 @@
 				self.importSetButtons();
 			}, 'text');
 		},
+		updateCachedUserSets: function (format) {
+			if(this.userSets?.[format]) return;
+
+			this.userSets = this.userSets || {};
+			this.userSets[format] = this.userSets[format] || {};
+
+			var duplicateNameIndices = {};
+			for(const team of teams) {
+				if(team.format === format && team.capacity === 24) {
+					const setList = Storage.unpackTeam(team.team);
+					for(const pokemon of setList) {
+						var name = pokemon.name + " " + (duplicateNameIndices[pokemon.name] || "");
+						var sets = this.userSets[format][pokemon.species];
+						this.userSets[format][pokemon.species] = $.extend(sets || {}, {[name] : pokemon});
```

Comment:
```suggestion
						var sets = this.userSets[format][pokemon.species] || {};
						sets[name] = pokemon;
						this.userSets[format][pokemon.species] = sets;
```

I think this is what you're trying to do?

Reply:
Exactly, thanks! :)